sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalogSuite.scala

```diff
@@ -450,6 +451,26 @@ abstract class SessionCatalogSuite extends PlanTest {
     }
   }
 
+  test("alter table add columns") {
+    withBasicCatalog { sessionCatalog =>
+      sessionCatalog.createTable(newTable("t1", "default"), ignoreIfExists = false)
+      val oldTab = sessionCatalog.externalCatalog.getTable("default", "t1")
+      sessionCatalog.alterTableSchema(TableIdentifier("t1", Some("default")),
+        oldTab.schema.add("c3", IntegerType))
+      val newTab = sessionCatalog.externalCatalog.getTable("default", "t1")
+      if (sessionCatalog.externalCatalog.isInstanceOf[InMemoryCatalog]) {
+        assert(newTab.schema.toString == oldTab.schema.add("c3", IntegerType).toString)
+      } else {
+        // HiveExternalCatalog will always arrange the partition columns to the end
```

Comment:
We need to do this(arrange the partition columns to the end) in `SessionCatalog.alterTableSchema`.